src/array.cr

```diff
@@ -1770,4 +1770,12 @@ class Array(T)
       end
     end
   end
+
+  def bsearch
+    idx=(0...self.size).bsearch{|i|yield self[i]}
+    idx ? self[idx] : nil
```

Comment:
Uh, in https://github.com/cielavenir/crystal-bsearch/blob/master/lib/bsearch.cr , Array#bsearch is defined after Range#bsearch. We might need to define Array#bsearch in range.cr...